test/byContract/registries/resources.test.ts

```diff
@@ -0,0 +1,224 @@
+        const ensRegistryAddress = (await deployments.get('ENSRegistry')).address;
+        ensRegistry = ENSRegistry__factory.connect(ensRegistryAddress, anyone);
+
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+
+        appRegistry = AppRegistry__factory.connect(await iexecPoco.appregistry(), anyone);
+        datasetRegistry = DatasetRegistry__factory.connect(
+            await iexecPoco.datasetregistry(),
+            anyone,
+        );
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            await iexecPoco.workerpoolregistry(),
+            anyone,
+        );
+    }
+
+    describe('App', () => {
+        const createAppArgs = [
+            `App`,
+            'DOCKER',
+            MULTIADDR_BYTES,
+            ethers.utils.id(`Content of my app`),
+            '0x1234',
+        ] as [string, string, BytesLike, BytesLike, BytesLike];
+
+        beforeEach(async () => {
+            const appAddress = await appRegistry.callStatic.createApp(
+                appProvider.address,
+                ...createAppArgs,
+            );
+            await appRegistry
+                .createApp(appProvider.address, ...createAppArgs)
+                .then((tx) => tx.wait());
+            app = App__factory.connect(appAddress, anyone);
+        });
+
+        it('Should create an app and verify its details', async () => {
+            expect(await app.registry()).to.equal(appRegistry.address);
+            expect(await app.owner()).to.equal(appProvider.address);
+            expect(await app.m_appName()).to.equal(createAppArgs[0]);
+            expect(await app.m_appType()).to.equal(createAppArgs[1]);
+            expect(await app.m_appMultiaddr()).to.equal(createAppArgs[2]);
+            expect(await app.m_appChecksum()).to.equal(createAppArgs[3]);
+            expect(await app.m_appMREnclave()).to.equal(createAppArgs[4]);
+        });
+
+        it('Should set the ENS name for the app', async () => {
+            const newENSName = 'myApp.eth';
+            const reverseRootNameHash = ethers.utils.namehash('addr.reverse');
+            const reverseRegistrarAddress = await ensRegistry.owner(reverseRootNameHash);
+            const reverseResolverAddress = await ReverseRegistrar__factory.connect(
+                reverseRegistrarAddress,
+                anyone,
+            ).defaultResolver();
+            const reverseResolver = PublicResolver__factory.connect(reverseResolverAddress, anyone);
+
+            await app
+                .connect(appProvider)
+                .setName(ensRegistry.address, newENSName)
+                .then((tx) => tx.wait());
+
+            const nameHash = ethers.utils.namehash(`${app.address.substring(2)}.addr.reverse`);
+            expect(await reverseResolver.name(nameHash)).to.equal(newENSName);
+        });
+
+        it('Should not reinitialize created app', async () => {
```

Comment:
You can move that below "Should create an <app/dataset/pool> and verify its details" since it is related to the creation of the <app/dataset/pool>